shell/platform/darwin/macos/framework/Source/FlutterAccessibilityMac.mm

```diff
@@ -0,0 +1,575 @@
+    case ax::AXEventGenerator::Event::SELECTED_CHANGED:
+    case ax::AXEventGenerator::Event::SELECTION_IN_TEXT_FIELD_CHANGED:
+    case ax::AXEventGenerator::Event::SET_SIZE_CHANGED:
+    case ax::AXEventGenerator::Event::SORT_CHANGED:
+    case ax::AXEventGenerator::Event::STATE_CHANGED:
+    case ax::AXEventGenerator::Event::TEXT_ATTRIBUTE_CHANGED:
+    case ax::AXEventGenerator::Event::RANGE_VALUE_MAX_CHANGED:
+    case ax::AXEventGenerator::Event::RANGE_VALUE_MIN_CHANGED:
+    case ax::AXEventGenerator::Event::RANGE_VALUE_STEP_CHANGED:
+    case ax::AXEventGenerator::Event::WIN_IACCESSIBLE_STATE_CHANGED:
+      // There are some notifications that aren't meaningful on Mac.
+      // It's okay to skip them.
+      return;
+  }
+  FireNativeMacNotification(native_node, mac_notification);
+}
+
+void FlutterAccessibilityMac::FireNativeMacNotification(gfx::NativeViewAccessible native_node,
+                                                        NSString* mac_notification) {
+  FML_DCHECK(mac_notification);
+  // FML_DCHECK(native_node);
+  NSAccessibilityPostNotification(native_node, mac_notification);
+}
+
+void FlutterAccessibilityMac::FireNativeMacNotificationWithUserInfo(
+    gfx::NativeViewAccessible native_node,
+    NSString* mac_notification,
+    NSDictionary* user_info) {
+  FML_DCHECK(mac_notification);
+  FML_DCHECK(native_node);
+  FML_DCHECK(user_info);
+  NSAccessibilityPostNotificationWithUserInfo(native_node, mac_notification, user_info);
+}
+
+bool FlutterAccessibilityMac::IsInGeneratedEventBatch(
+    ax::AXEventGenerator::Event event_type) const {
+  for (const auto& event : *(GetBridge()->GetEventGenerator())) {
+    if (event.event_params.event == event_type)
+      return true;  // Any side effects will have already been handled.
+  }
+  return false;
+}
+
+// FlutterAccessibilityMac override
+void FlutterAccessibilityMac::DispatchAccessibilityAction(uint16_t target,
+                                                          FlutterSemanticsAction action,
+                                                          uint8_t* data,
+                                                          size_t data_size) {
+  [GetFlutterEngine() dispatchSemanticsAction:target action:action data:data dataSize:data_size];
+}
+
+gfx::NativeViewAccessible FlutterAccessibilityMac::GetNativeViewAccessible() {
+  FML_DCHECK(ax_platform_node_);
+  return ax_platform_node_->GetNativeViewAccessible();
+}
+
+gfx::NativeViewAccessible FlutterAccessibilityMac::GetParent() {
+  gfx::NativeViewAccessible parent = FlutterAccessibility::GetParent();
+  if (!parent) {
+    return GetFlutterEngine().viewController.view;
+  }
+  return parent;
+}
+
+SkRect FlutterAccessibilityMac::GetBoundsRect(const AXCoordinateSystem coordinate_system,
+                                              const AXClippingBehavior clipping_behavior,
+                                              AXOffscreenResult* offscreen_result) const {
+  // TODO: consider screen dpr.
```

Comment:
If this is a real TODO, file an issue.

Is it really necessary to consider the DPR here though? I'm not quite clear on which subsystems are DPR aware or not here. It may help to explain that in the TODO.